Mage/src/main/java/mage/abilities/common/TargetOfOpponentsSpellOrAbilityTriggeredAbility.java

```diff
@@ -0,0 +1,91 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * "Whenever you or a permanent you control becomes the target of a spell or ability an opponent controls,"
+ *  AND
+ *  "Whenever you become the target of a spell or ability an opponent controls,"
+ *
+ * @author Alex-Vasile
+ */
+public class TargetOfOpponentsSpellOrAbilityTriggeredAbility extends TriggeredAbilityImpl {
+
+    private Boolean onlyController = Boolean.FALSE;
+
+    public TargetOfOpponentsSpellOrAbilityTriggeredAbility(Effect effect) {
+        this(effect, false);
+    }
+
+    // NOTE: Using Boolean instead of boolean in order to have a second constructor with (Effect, "boolean") signature
+    public TargetOfOpponentsSpellOrAbilityTriggeredAbility(Effect effect, Boolean onlyController) {
```

Comment:
Please, *please*, **PLEASE** don't do this. I just finished ranting on Discord about how much I hate constructors with multiple boolean parameters, and this is somehow even worse.

Reply:
As far as I can tell, I need a second boolean for when when the target has to be "you" rather than "you or a permanent you control".

I must have missed the rant, but I very must dislike the current approach we take to constructors, and the need to use Boolean instead of boolean is part of that.

Reply:
Thoughts?

Reply:
I would prefer that this and `BecomesTargetControlledPermanentTriggeredAbility` be merged into a single class for "Whenever {a} becomes the target of a {b}" that takes filters for both {a} and {b}, but that's a refactor I can do myself once this is merged.

Reply:
> I would prefer that this and `BecomesTargetControlledPermanentTriggeredAbility` be merged into a single class for "Whenever {a} becomes the target of a {b}" that takes filters for both {a} and {b}, but that's a refactor I can do myself once this is merged.

Sounds good, I will mark this as resolved and add a TODO in the main comment of this PR.